CONTRIBUTING.md

```diff
@@ -1,41 +1,223 @@
+
+```bash
+rush rebuild
+```
+
+### Cleaning Rush Js Cash
+```bash
+rush purge
+```
+
+## Start the Development
+
+The typical development process involves:
+
+1. Develop or modify the package.
+2. Build the packages.
+3. Clean Rush cache.
+4. Navigate to the corresponding example folder.
+5. Install example project dependencies with ``pnpm i``.
+6. Run the example with ``pnpm dev`` or ``pnpm serve`` (check the case on package.json).
+7. Test your changes thoroughly.
+
+## Adding E2E Tests
+
+1. Install e2e project dependencies with the command:
+    ```bash
+    pnpm i
+    ```
+2. Add necessary IDs on HTML Tags to the example project
+3. Write the tests in a ``*.spec.ts`` on [e2e/tests](./e2e/tests) folder
+4. Clean examples and rushJs using the command:
+   - Unix-based system:
+    ```bash
+   pnpm ex:clean:unix
+    ```
+   - Windows:
+   ```bash
+   pnpm ex:clean:windows
+   ```
+5. Install examples dependencies:
+   - Unix-based system:
+    ```bash
+   pnpm ex:install:unix
     ```
-2. Run the setup script (equal to `npm i` on each of the packages)
-    ```sh
-    rush update
+   - Windows:
+   ```bash
+   pnpm ex:install:windows
+   ```
+6. Build examples projects:
+   - Unix-based system:
+    ```bash
+   pnpm ex:install:unix
     ```
+   - Windows:
+   ```bash
+   pnpm ex:build:windows
+   ```
+7. Run the E2E tests:
+   - Opening browser:
+    ```bash
+   pnpm test
+    ```
+   - Headless:
+   ```bash
+   pnpm test:headless
+   ```
 
-## Build (equals to `npm run build` on each of the packages)
-```sh
-rush rebuild
+## Formatting code
+
+For code formatting, use the following commands:
+
+- Validate code formatting:
+```bash
+rush lint
 ```
 
-## Before commit
-Build the project and run:
-```sh
-rush change
+- Format the code according to standards:
+```bash
+rush format
 ```
 
-## Publish
-- Build the project
-- Increase the version of each package
-- Commit
-- Go to each of the packages and run:
-    ```sh
-    npm publish
+> Due to limitations in RushJs, Prettier has been left in the auto-installer of RushJs. However, ESLint has been configured individually within each package. In case an error occurs stating "eslint is not recognized as a command" while running any of the commands above, it is necessary to install the dependencies of the packages to resolve this issue.
+
+## GitHooks
+
+- In the [``pre-commit``](./common/git-hooks/pre-commit) hook, code formatting is checked.
+
+- In the [``pre-push``](./common/git-hooks/pre-push) hook, it's verified whether the developer has added a changelog message for their changes.
+
+To add a changelog message on your commit/push, follow these steps:
+
+1. **Add Your Changes to Commit:**
+
+   Incorporate your modifications into a commit using a GIT GUI or execute the following command:
+    ```bash
+   git add .
+   ```
+2. **Run the Rush Change Command:**
+
+   Use the following command, providing a descriptive message of your alterations, and select the appropriate type of modification:
```

Comment:
It's important to say that if there isn't changes relevant for the users there is no need to add a change message.